scripts/build.py

```diff
@@ -805,49 +805,49 @@ def archive_engine(self):
         if self.target_platform in ['x86_64-linux', 'x86_64-darwin', 'x86_64-win32']:
             launcher_name = format_exes("launcher", self.target_platform)[0]
             launcherbin = join(bin_dir, launcher_name)
-            self.upload_file(launcherbin, '%s/%s' % (full_archive_path, launcher_name))
+            self.upload_to_archive(launcherbin, '%s/%s' % (full_archive_path, launcher_name))
```

Comment:
The `upload_to_archive()` function will upload to an S3 path based on the current channel and in a path format that we can easily filter and give different expiration policies:

d.defold.com/beta/files/sha1/*
d.defold.com/stable/files/sha1/*
d.defold.com/alpha/files/sha1/*
...

The upload_to_archive() function will also handle setup of redirects to the old formation (d.defold.com/archive/sha1/*)